lib/langchain/message.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module Langchain
+  class Message
+    attr_reader :content, :additional_kwargs
+
+    def initialize(content, additional_kwargs = nil)
+      @content = content
+      @additional_kwargs = additional_kwargs
+    end
+
+    def type
+      raise NotImplementedError
+    end
+
+    def to_s
+      content
+    end
+
+    def ==(other)
+      to_json == other.to_json
+    end
+
+    def to_json(options = {})
+      hash = {
+        content: content
+      }
+
+      hash[:additional_kwargs] = additional_kwargs unless additional_kwargs.nil? || additional_kwargs.empty?
+
+      hash.to_json
+    end
+  end
+
+  class AIMessage < Message
+    def type
+      "ai"
+    end
+  end
+
+  class HumanMessage < Message
+    def type
+      "human"
+    end
+  end
+
+  class SystemMessage < Message
```

Comment:
maybe also a new message that represents a function execution, wdyt? (not used yet, so it can be on a new pr when its used):
```
FunctionMessage < Message
  def type
    "system"
  end
end
```

Reply:
I'd prefer to add it on the next PR so each PR is self-contained and we never have unused code